src/action_library.js

```diff
@@ -0,0 +1,67 @@
+const core = require('@actions/core');
+const fs = require('fs');
+const { env } = require('process');
+const yaml = require('yaml');
+
+class action_library {
+  constructor() {
+    this.currentEnvironmentPath = process.env['GITHUB_ENV'];
+  }
+
+  getFileYaml = path => {
+    let fileData = fs.readFileSync(path, 'utf8');
+    let fileYaml = yaml.parse(fileData);
+    return fileYaml;
+  };
+
+  filterObjectByKeyPart = (filterObject, filterFunction, keyTransform) =>
+    Object.keys(filterObject)
+      .filter(k => filterFunction(k))
+      .reduce((scoped, key) => {
+        let returnKey = keyTransform ? keyTransform(key) : key;
+        return {
+          ...scoped,
+          [returnKey]: filterObject[key]
+        };
+      }, {});
+
+  getCurrentEnvironmentVars = () => this.filterObjectByKeyPart(process.env, k => k.includes('@'));
+
+  keyName = key => key.split('@')[0];
+
+  keyIsPartOfScope = (scope, key) => {
+    let scopes = key.split('@')[1].split(' ');
+    return scopes.includes(scope);
+  };
+
+  buildEnvironmentDictionary = (scope, input, environment) => {
+    let environmentDictionary = {};
+    let inputScoped = this.filterObjectByKeyPart(input, k => this.keyIsPartOfScope(scope, k), this.keyName);
```

Comment:
I'm curious how this is working. `filterObjectByKeyPart`'s third parameter should be a function based on how it's used, but `this.keyName` is a string...am I missing something super obvious?

Reply:
...I was missing something super obvious 😆. This isn't the first time I've made a mistake similar to this either. I'm not sure what it is about javascript function properties that are one liners like this, but I space that you actually need to invoke them to get the value 🤦‍♂️